networkx/algorithms/distance_measures.py

```diff
@@ -163,7 +104,20 @@ def _extrema_bounding(G, compute="diameter"):
         high = not high
 
         # get distances from/to current node and derive eccentricity
-        dist = dict(nx.single_source_shortest_path_length(G, current))
+        dist = nx.single_source_bellman_ford_path_length(
```

Comment:
I see what you mean. How about I switch to `shortest_path_length` in `_extrema_bounding` and provide a distance_method arg that defaults to "dijkstra" in `_extrema_bounding`, `diameter`, `periphery`, `radius`, and `center`?

Reply:
Upon thinking about this more...  This is the `distance_method` module. The edge weights are distances. So it doesn't make sense for them to be negative. So, we should just use dijkstra for the weighted case.  That is the default for `shortest_path_length`.  So I suggest using `shortest_path_length for all the functions -- and not add an extra arg to all those functions.   

Thoughts?

Reply:
Hm, I'm not sure. I'll defer to your guidance, but since we've given the user the freedom to use any weight function, I feel like we should be able to handle negative weights.

If you think the distance_method arg cluters the api, what if I just check to see if weights are negative in `_extrema_bounding` and `eccentricity`?

Reply:
The user can provide an arbitrary function, but it still has to make sense... Just like we let them provide any weight value, but it has to make sense.  And you shouldn't have to check for negative weights. That will be done in the `shortest_path_length`.  But we should probably add another phrase in the weight doc_string description that the weights should be positive (since they are distances).

Reply:
Ok, this makes sense. I've removed the distance_method argument and added a sentence to the relevant docstrings indicating that weights should be positive.